src/magnet_uri.cpp

```diff
@@ -215,8 +215,12 @@ namespace libtorrent {
 			url = uri.substr(pos, find(uri, "&", pos) - pos);
 		}
 
-		string_view btih = url_has_argument(uri, "xt");
-		if (btih.empty())
+		string_view btih = unescape_string(url_has_argument(uri, "xt"), ec);
```

Comment:
by the way, assigning a temporary to the string view is what's causing all the failures in CI